test/e2e/test_catalogue_item.py

```diff
@@ -2,2198 +2,1539 @@
+        """
+
+        self._get_response = self.test_client.get(f"/v1/catalogue-items/{catalogue_item_id}")
+
+    def check_get_catalogue_item_success(self, expected_catalogue_item_get_data: dict) -> None:
+        """
+        Checks that a prior call to `get_catalogue_item` gave a successful response with the expected data returned.
+
+        :param expected_catalogue_item_get_data: Dictionary containing the expected system data returned as would be
+                                                 required for a `CatalogueItemSchema`. Does not need mandatory IDs (e.g.
+                                                 manufacturer_id) as they will be added automatically to check they are
+                                                 as expected.
+        """
+
+        assert self._get_response.status_code == 200
+        assert self._get_response.json() == self.add_ids_to_expected_catalogue_item_get_data(
+            expected_catalogue_item_get_data
+        )
+
+    def check_get_catalogue_item_failed_with_detail(self, status_code: int, detail: str) -> None:
+        """
+        Checks that a prior call to `get_catalogue_item` gave a failed response with the expected code and error
+        message.
+
+        :param status_code: Expected status code of the response.
+        :param detail: Expected detail given in the response.
+        """
+
+        assert self._get_response.status_code == status_code
+        assert self._get_response.json()["detail"] == detail
+
+
+class TestGet(GetDSL):
+    """Tests for getting a catalogue item."""
+
+    def test_get(self):
+        """Test getting a catalogue item."""
+
+        self.post_catalogue_category(CATALOGUE_CATEGORY_POST_DATA_LEAF_NO_PARENT_NO_PROPERTIES)
+        self.post_manufacturer(MANUFACTURER_POST_DATA_REQUIRED_VALUES_ONLY)
+        catalogue_item_id = self.post_catalogue_item(CATALOGUE_ITEM_DATA_REQUIRED_VALUES_ONLY)
+
+        self.get_catalogue_item(catalogue_item_id)
+        self.check_get_catalogue_item_success(CATALOGUE_ITEM_GET_DATA_REQUIRED_VALUES_ONLY)
+
+    def test_get_with_non_existent_id(self):
+        """Test getting a catalogue item with a non-existent ID."""
+
+        self.get_catalogue_item(str(ObjectId()))
+        self.check_get_catalogue_item_failed_with_detail(404, "Catalogue item not found")
+
+    def test_get_with_invalid_id(self):
+        """Test getting a catalogue item with an invalid ID."""
+
+        self.get_catalogue_item("invalid-id")
+        self.check_get_catalogue_item_failed_with_detail(404, "Catalogue item not found")
+
+
+class ListDSL(GetDSL):
+    """Base class for list tests."""
+
+    def get_catalogue_items(self, filters: dict) -> None:
+        """
+        Gets a list catalogue items with the given filters.
+
+        :param filters: Filters to use in the request.
+        """
+
+        self._get_response = self.test_client.get("/v1/catalogue-items", params=filters)
+
+    def post_test_catalogue_items(self) -> list[dict]:
```

Comment:
Is `test` needed in the name of this method?

Reply:
It matches the previous e2e tests as I thought `post_catalogue_items` is a bit confusing as it doesnt post something its given, although I admit the other names tend to be clearer e.g. `post_test_catalogue_category_with_child` so open to a better name. The only thing I can think of right now is just being more explicit e.g. `post_two_test_catalogue_items`.

Reply:
Oh I see what it was supposed to mean, I just thought it was a typo. Please ignore me and keep it as it is, sorry.